File: backend/src/models/respondent.ts
import {
  ModelType,
  prop,
  staticMethod,
  Typegoose,
} from "typegoose";

enum RespondentType {
  Administrator = "Administrator",
  Official = "Official",
  Executive = "Executive",
  Monitor = "Monitor",
  Citizen = "Citizen",
}
class Respondent extends Typegoose {

  //
  @staticMethod
  public static findByUserID(
    this: ModelType<Respondent> & typeof Respondent,
    userID: string,
  ) {
    return this.findOne({ userID });
  }
  @staticMethod
  public static findByUserType(
    this: ModelType<Respondent> & typeof Respondent,
    userType: string,
  ) {
    return this.find({ userType });
  }
  //
  @prop({ required: true, trim: true })
  public firstName!: string;
  //
  @prop({ required: true, trim: true })
  public email!: string;
  //
  @prop({ trim: true })
  public cellphone!: string;
  //
  @prop({ required: true, trim: true })
  public lastName!: string;
  //
  @prop({ index: true, trim: true })
  public respondentID!: string;
  
  @prop({ trim: true })
  public fcmToken?: string;
  //
  @prop({ trim: true })
  public gender?: string;
  //
  
  @prop({ required: true })
  public respondentType!: RespondentType;
  //
  @prop({ required: true, default: new Date().toISOString() })
  public created!: string;
}

export default Respondent;

File: monitorlibrary/lib/data/respondent.dart
import 'package:meta/meta.dart';

class Respondent {
  String? firstName, lastName, userId, email, cellphone, gender;

  Respondent({required this.firstName, required this.lastName,  this.email, this.cellphone,
    required this.gender,  });

  Respondent.fromJson(Map data) {
    this.firstName = data['firstName'];
    this.lastName = data['lastName'];
    this.userId = data['userId'];
    this.email = data['email'];
    this.cellphone = data['cellphone'];
    this.gender = data['gender'];
  }
  Map<String, dynamic> toJson() {

    Map<String, dynamic> map = {
      'firstName': firstName,
      'lastName': lastName,
      'userId': userId,
      'email': email,
      'cellphone': cellphone,
      'gender': gender,
    };
    return map;
  }

}